### DISCUSSION TITLE
Certain Pictures Not Loading With Dark Reader

### DISCUSSION THREAD

**Discussion by HDMOVIESOURCE** [darkreader/darkreader#11193] [General]
I have a few websites that don't load their logo when Dark Reader is active, do you know why this is?

**Comment by ngoclong19** [comment]
Could you provide some example websites?

**Comment by HDMOVIESOURCE** [comment]
https://www.hdmoviesource.com/
The logo doesn't show on every page. When dark reader is off all logo images load perfectly.

**Reply by ngoclong19** [reply]
Below is the analysis for my reference purpose, please skip to the last section if not interested.

---

When DR is enable, the website loads this image, which returns 404 error (\*):
`https://www.hdmoviesource.com/images/template/header_bg.png`

When DR is not enable, the website load this image (\*\*):
`https://www.hdmoviesource.com/v/vspfiles/templates/Supply/images/template/header_bg.png`

From this file, in the function which contains `a = $('meta[itemprop="name"] + a')`:
`https://www.hdmoviesource.com/v/vspfiles/templates/Supply/js/min/template.min.js`
The website will take the background image of `<a class="vol-logo__link">` element, then apply it to the image source of `<img class="logo__img img-responsive">` element (\*\*\*).

The background image of `<a>` element is defined here as `url("../images/template/header_bg.png")` (\*\*\*\*):
`https://www.hdmoviesource.com/v/vspfiles/templates/Supply/css/template.css`

My understanding of DR's working is as followed, please take this with a grain of salt:
- Inject the `background-image` rule directly into the page, with the value taken as is from CSS.
Here is the catch, the value at (\*\*\*\*) is a relative path, the browser will resolve it to a different URL (\*) instead of (\*\*).
I think the website applies (\*\*\*) during this time, leads to incorrect image URL.
- Invert the background image using the original browser resolved URL (\*\*), and replace the injected value above with the inverted one.

---

If you can make changes to the website, I suggest watch the `html` element for attribute `data-darkreader-scheme="dark"` and apply (\*\*\*) again.

Or add these CSS rules to `<a>` element, `display: block; height: 150px; background-size: initial;`.
This way, you don't need the nested `<img>` element.

**Reply by HDMOVIESOURCE** [reply]
Thank you, spoke to my devs and I believe this issue should now be fixed, thank you for the detailed explanation.
